src/mcp/client/stdio/__init__.py

```diff
@@ -159,4 +169,48 @@ async def stdin_writer():
     ):
         tg.start_soon(stdout_reader)
         tg.start_soon(stdin_writer)
-        yield read_stream, write_stream
+        try:
+            yield read_stream, write_stream
+        finally:
+            # Clean up process to prevent any dangling orphaned processes
+            if sys.platform == "win32":
+                await terminate_windows_process(process)
+            else:
+                process.terminate()
```

Comment:
This still hangs on linux if the invoked process expects SIGINT, eg: https://github.com/modelcontextprotocol/servers/blob/main/src/everything/index.ts#L13

Shouldn't we try to force kill, same as we do on Windows?

Reply:
@dsp-ant check this out please: https://github.com/modelcontextprotocol/python-sdk/pull/555